kin/blockchain/channel_manager.py

```diff
@@ -1,77 +1,143 @@
+    def __init__(self, channel_seeds):
+        """
+        Crete a channel manager instance
+        :param list[str] channel_seeds: The seeds of the channels to use
+        """
+        self.channel_pool = ChannelPool(channel_seeds)
 
-        :param str memo_text: (optional) a text to add as transaction memo.
+    @contextmanager
+    def get_channel(self, timeout=CHANNEL_GET_TIMEOUT):
+        """
+        Get an available channel
+        :param float timeout: (Optional) How long to wait before raising an exception
+        :return a free channel seed
+        :rtype str
 
-        :return: transaction object
-        :rtype: dict
+        :raises KinErrors.ChannelBusyError
         """
-        # get an available channel builder first (blocking with timeout)
         try:
-            builder = self.channel_builders.get(True, CHANNEL_QUEUE_TIMEOUT)
+            channel = self.channel_pool.get(timeout=timeout)
         except queue.Empty:
-            raise ChannelsBusyError
+            raise ChannelsBusyError()
 
-        # operation source is always the base account
-        source = self.base_address if builder.address != self.base_address else None
+        try:
+            yield channel
+        finally:
+            if self.channel_pool.queue[channel] != ChannelStatuses.UNDERFUNDED:
+                self.put_channel(channel)
+
+    def put_channel(self, channel, timeout=CHANNEL_PUT_TIMEOUT):
+        """
+        Set a channel status back to FREE
+        :param str channel: the channel to set back to FREE
+        :param float timeout: (Optional) How long to wait before raising an exception
 
-        # add operation (using external partial) and sign
+        :raises KinErrors.ChannelsFullError
+        """
         try:
-            add_ops_fn(builder)(source=source)
-            # update the previous fee that was only used for initialization
-            builder.fee = fee
-            if memo_text:
-                builder.add_text_memo(memo_text)  # max memo length is 28
-
-            builder.sign()  # always sign with a channel key
-            if source:
-                builder.sign(secret=self.base_key)  # sign with the base key if needed
-        except:
-            # If something fails when building the tx, clear and return the builder to queue.
-            builder.clear()
-            self.channel_builders.put(builder,timeout=CHANNEL_PUT_TIMEOUT)
-            raise
-        return builder
+            self.channel_pool.put(channel, timeout=timeout)
+        except queue.Full:
+            raise ChannelsFullError()
+
+    def get_status(self, verbose=False):
+        """
+        Return the current status of the channel manager
+        :param bool verbose: Include all channel seeds and their statuses in the response
+        :return: dict
+        """
+        free_channels = len(self.channel_pool.get_free_channels())
+        status = {
+            'total_channels': len(self.channel_pool.queue),
```

Comment:
if you're pre calculating `free_channels` to keep DRY, you might as well do the same to `total_channels`